### PR TITLE
Add num256 to num casting (in function signature)

### PR DESCRIPTION
**Pull request by jacqueswww**
### - What I did

- Altered the max values that get stored in memory for clamps to compare against.
- Added support for num(num256) function signatures.

### - How I did it

- Edited base_type_conversion (Thanks @DavidKnott !)
- Applied changes in BaseType and canonicalize_type to generate the appropriate ABI signature.

### - How to verify it

Check tests and play with function using the `num(num256)` signature:

```
def _num256_to_num(x: num(num256)) -> num:
    return x
```

### - Description for the changelog
Added support for num(num256) function signatures.

### CODE REVIEW

File: viper/parser.py [hunk 1]
```diff
@@ -230,10 +230,10 @@ def get_contracts_and_defs_and_globals(code):
...
                                         ['mstore', 28, ['calldataload', 0]],
                                         ['mstore', ADDRSIZE_POS, 2**160],
-                                        ['mstore', MAXNUM_POS, 2**128 - 1],
-                                        ['mstore', MINNUM_POS, -2**128 + 1],
-                                        ['mstore', MAXDECIMAL_POS, (2**128 - 1) * DECIMAL_DIVISOR],
-                                        ['mstore', MINDECIMAL_POS, (-2**128 + 1) * DECIMAL_DIVISOR],
+                                        ['mstore', MAXNUM_POS, 2**127 - 1],
+                                        ['mstore', MINNUM_POS, -2**127],
+                                        ['mstore', MAXDECIMAL_POS, (2**127 - 1) * DECIMAL_DIVISOR],
+                                        ['mstore', MINDECIMAL_POS, (-2**127 + 1) * DECIMAL_DIVISOR],
```
Comment (fubuloubu) [inline viper/parser.py]: Just `(-2**127)`

Comment (jacqueswww) [reply viper/parser.py]: Ha, thanks ;)

Comment (jacqueswww) [reply viper/parser.py]: Done :)

**Comment by DavidKnott** [comment]
@jacqueswww LGTM.

**Comment by fubuloubu** [comment]
Yoda says "me looks good to, merge in you must"

**Comment by DavidKnott** [comment]
@jacqueswww Just wanted time to pull it down and play with it to be 100% sure :)

**Comment by jacqueswww** [comment]
@DavidKnott sweet thanks ! :)

**Pull request merged**